prysm/x/polarization.py

```diff
@@ -364,3 +454,100 @@ def pauli_coefficients(jones):
     c3 = 1j*(jones[..., 0, 1] - jones[..., 1, 0]) / 2
 
     return c0, c1, c2, c3
+
+
+def jones_adapter(prop_func):
+    """wrapper around prysm.propagation functions to support polarized field propagation
+
+    Parameters
+    ----------
+    prop_func : callable
+        propagation function to decorate
+
+    Notes
+    -----
+    There isn't anything particularly special about polarized field propagation. We simply 
+    leverage the independence of the 4 "polarized" components of an optical system expressed
+    as a Jones matrix
+
+    J = [
+        [J00,J01],
+        [J10,J11]
+    ]
+
+    The elements of this matrix can be propagated as incoherent wavefronts to express the polarized
+    response of an optical system. All `jones_adapter` does is call a given propagation function
+    4 times, one for each element of the Jones matrix.
+
+    Returns
+    -------
+    callable
+        decorated propagation function
+    """
+
+    @functools.wraps(prop_func)
+    def wrapper(*args,**kwargs):
+
+        
+        # this is a function
+        wavefunction = args[0]
+        if len(args) > 1:
+            other_args = args[1:]
+        else:
+            other_args = ()
+
+        if wavefunction.ndim == 2:
+            # pass through non-jones case
+            return prop_func(*args,**kwargs)
+
+        J00 = wavefunction[...,0,0]
+        J01 = wavefunction[...,0,1]
+        J10 = wavefunction[...,1,0]
+        J11 = wavefunction[...,1,1]
+        tmp = []
+        for E in [J00, J01, J10, J11]:
+            ret = prop_func(E, *other_args, **kwargs)
+            tmp.append(ret)
+        
+        out = np.empty([*ret.shape,2,2],dtype=ret.dtype)
+        out[...,0,0] = tmp[0]
+        out[...,0,1] = tmp[1]
+        out[...,1,0] = tmp[2]
+        out[...,1,1] = tmp[3]
+        
+        return out
+    
+    return wrapper
+
+def add_jones_propagation(funcs_to_change=supported_propagation_funcs):
+    """apply decorator to supported propagation functions
+
+    Parameters
+    ----------
+    funcs_to_change : list, optional
+        list of propagation functions to add polarized field propagation to, by default supported_propagation_funcs
+    """
+
+    for name,func in vars(propagation).items():
+        if name in funcs_to_change:
+            setattr(propagation, name, jones_adapter(func))
+
+def apply_polarization_to_field(field):
```

Comment:
This is not used in the module and does not have an intuitive mapping between its name and what it does; please remove

Reply:
From the `Polarized_Propagation.ipynb` demo:
<img width="405" alt="image" src="https://github.com/brandondube/prysm/assets/25557892/7ac6d617-daa5-4436-aef8-396d25b80e22">

This was added in response to a comment on #91 regarding the demo containing `A = A[…, np.newaxis, np.newaxis]` to enable element-wise multiplication. Would you prefer `apply_polarization_component_to_field` or something similar?

Reply:
It is a design question, I think

I understand that `vvr * A` is a very nice syntax since it reads like the calculus

What I meant was a function ~=

```
def apply_polarization_optic(field, pol):
    # internal logic understands 2D -> 4D broadcasts
```

But this is fairly verbose if it is going to be used all the time

Perhaps instead the routines that create polarization components could return wavefronts, and the math overrides on wavefronts could be extended to include the newaxis's when there is a dimensionality mismatch?

Reply:
Added in 4dbd050

Reply:
The change in `4dbd050` is not quite right; in the event that the field is already polarization compatible, it will make it `(N,M,2,2,1,1)`; please add an `if field.ndim == 2` check (or != 4, up to you)

Reply:
change made in b885996